code/modules/mob/living/living.dm

```diff
@@ -648,66 +622,98 @@
 		..(pressure_difference, direction, pressure_resistance_prob_delta)
 
 /mob/living/can_resist()
-	return !((next_move > world.time) || incapacitated(ignore_restraints = TRUE))
+	return !((next_move > world.time) || !CHECK_MOBILITY(src, MOBILITY_RESIST))
 
+/// Resist verb for attempting to get out of whatever is restraining your motion. Gives you resist clickdelay if do_resist() returns true.
 /mob/living/verb/resist()
 	set name = "Resist"
 	set category = "IC"
 
 	if(!can_resist())
 		return
-	changeNext_move(CLICK_CD_RESIST)
 
+	if(do_resist())
+		changeNext_move(CLICK_CD_RESIST)
+
+/// The actual proc for resisting. Return TRUE to give clickdelay.
+/mob/living/proc/do_resist()
 	SEND_SIGNAL(src, COMSIG_LIVING_RESIST, src)
 	//resisting grabs (as if it helps anyone...)
-	if(!restrained(ignore_grab = 1) && pulledby)
-		visible_message("<span class='danger'>[src] resists against [pulledby]'s grip!</span>")
-		log_combat(src, pulledby, "resisted grab")
-		resist_grab()
-		return
-
-	//unbuckling yourself
+	// only works if you're not cuffed.
+	if(!restrained(ignore_grab = TRUE) && pulledby)
+		var/old_gs = pulledby.grab_state
+		attempt_resist_grab(FALSE)
+		// Return as we should only resist one thing at a time. Give clickdelay if the grab wasn't passive.
+		return old_gs? TRUE : FALSE
+
+	// unbuckling yourself. stops the chain if you try it.
 	if(buckled && last_special <= world.time)
-		resist_buckle()
+		log_combat(src, buckled, "resisted buckle")
+		return resist_buckle()
 
-	// CIT CHANGE - climbing out of a gut
-	if(attempt_vr(src,"vore_process_resist",args)) return TRUE
+	// CIT CHANGE - climbing out of a gut.
+	if(attempt_vr(src,"vore_process_resist",args))
+		//Sure, give clickdelay for anti spam. shouldn't be combat voring anyways.
+		return TRUE
 
 	//Breaking out of a container (Locker, sleeper, cryo...)
-	else if(isobj(loc))
+	if(isobj(loc))
 		var/obj/C = loc
 		C.container_resist(src)
+		// This shouldn't give clickdelays sometime (e.g. going out of a mech/unwelded and unlocked locker/disposals bin/etc) but there's so many overrides that I am not going to bother right now.
+		return TRUE
 
-	else if(canmove)
+	if(CHECK_BITFIELD(mobility_flags, MOBILITY_MOVE))
```

Comment:
CHECK_MOBILITY(src, MOBILITY_MOVE), and i think i missed another one like that earlier